umbraco/Umbraco-CMS
# v10 feature: Nullable reference types

# Notes
- Add nullabillity to the Core project
Some issues i ran into was nullable properties that was actually not nullable at all, just not initialized anywhere. Lets use UserSave as an example here: 
![image](https://user-images.githubusercontent.com/70372949/154478915-c55a970d-e634-4a51-8560-9929e502de4d.png)
We don't have a constructor for a UserSave, but we do use the json converter to Fill out the model when we get an incoming request, and with the attributes on "Username" we can see that it is required, and therefore it cannot be null. 
I've suppressed these nullable warnings for now using `= null!` but it is nonetheless interesting.

- the compiler doesn't think that string.IsNullOrEmpty() is a null check, so I had to suppress a lot of these warnings.
- Also the default ToString() method is now nullable, but it will only ever return null if you yourself overwrite it to do so, which means again a lot of these warnings just has to be suppressed with `!`
# How to test
- As this PR contains 925 files changes, this affects basically all of umbraco that uses core
- Run all the unit, integration & acceptance tests and they should pass
- Try creating a basic site with some document types & content
